app/render/job/shaderjob.h

```diff
@@ -87,6 +88,19 @@ class ShaderJob : public GenerateJob {
     interpolation_.insert(id, interp);
   }
 
+  void SetVertexCoordinates(QVector<float> vertex_coords)
+  {
+    vertex_overrides_.clear();  
+    foreach (float value, vertex_coords) {
+        vertex_overrides_.append(value);
+    }
+  }
+
+  QVector<float> GetVertexCoordinates()
```

Comment:
This too should be changed to `const QVector<float> &GetVertexCoordinates()` to prevent deep copies.